api/scanner/media_encoding/executable_worker/executable_worker.go

```diff
@@ -1,163 +1,50 @@
 
 import (
-	"fmt"
-	"io/ioutil"
 	"log"
-	"os"
-	"os/exec"
-	"strings"
 
 	"github.com/kkovaletp/photoview/api/utils"
-	"github.com/pkg/errors"
 	"gopkg.in/vansante/go-ffprobe.v2"
 )
 
 func InitializeExecutableWorkers() {
+	var err error
 	DarktableCli = newDarktableWorker()
-	FfmpegCli = newFfmpegWorker()
+	FfmpegCli, err = NewFfmpegWorker()
+	if err != nil {
+		log.Printf("An error occured when trying to initialize FFmpeg worker:\n%s", err)
+	}
 }
 
 var DarktableCli *DarktableWorker = nil
 var FfmpegCli *FfmpegWorker = nil
 
 type ExecutableWorker interface {
 	Path() string
-}
-
-type DarktableWorker struct {
-	path string
-}
-
-type FfmpegWorker struct {
-	path string
+	IsInstalled() bool
+	EncodeMp4(inputPath string, outputPath string) error
+	EncodeVideoThumbnail(inputPath string, outputPath string, probeData *ffprobe.ProbeData) error
 }
 
 func newDarktableWorker() *DarktableWorker {
 	if utils.EnvDisableRawProcessing.GetBool() {
 		log.Printf("Executable worker disabled (%s=1): darktable\n", utils.EnvDisableRawProcessing.GetName())
 		return nil
 	}
-
-	path, err := exec.LookPath("darktable-cli")
-	if err != nil {
-		log.Println("Executable worker not found: darktable")
-	} else {
-		version, err := exec.Command(path, "--version").Output()
-		if err != nil {
-			log.Printf("Error getting version of darktable: %s\n", err)
-			return nil
-		}
-
-		log.Printf("Found executable worker: darktable (%s)\n", strings.Split(string(version), "\n")[0])
-
-		return &DarktableWorker{
-			path: path,
-		}
-	}
-
-	return nil
+	return configureDarktableWorker()
 }
 
-func newFfmpegWorker() *FfmpegWorker {
+func NewFfmpegWorker() (*FfmpegWorker, error) {
 	if utils.EnvDisableVideoEncoding.GetBool() {
-		log.Printf("Executable worker disabled (%s=1): ffmpeg\n", utils.EnvDisableVideoEncoding.GetName())
-		return nil
-	}
-
-	path, err := exec.LookPath("ffmpeg")
-	if err != nil {
-		log.Println("Executable worker not found: ffmpeg")
-	} else {
-		version, err := exec.Command(path, "-version").Output()
-		if err != nil {
-			log.Printf("Error getting version of ffmpeg: %s\n", err)
-			return nil
-		}
-
-		log.Printf("Found executable worker: ffmpeg (%s)\n", strings.Split(string(version), "\n")[0])
-
-		return &FfmpegWorker{
-			path: path,
-		}
-	}
-
-	return nil
-}
-
-func (worker *DarktableWorker) IsInstalled() bool {
-	return worker != nil
-}
-
-func (worker *FfmpegWorker) IsInstalled() bool {
-	return worker != nil
-}
-
-func (worker *DarktableWorker) EncodeJpeg(inputPath string, outputPath string, jpegQuality int) error {
-	tmpDir, err := ioutil.TempDir("/tmp", "photoview-darktable")
-	if err != nil {
-		log.Fatal(err)
-	}
-	defer os.RemoveAll(tmpDir)
-
-	args := []string{
-		inputPath,
-		outputPath,
-		"--core",
-		"--conf",
-		fmt.Sprintf("plugins/imageio/format/jpeg/quality=%d", jpegQuality),
-		"--configdir",
-		tmpDir,
+		log.Printf("Executable worker disabled (%s=1): ffmpeg", utils.EnvDisableVideoEncoding.GetName())
+		return nil, nil
```

Comment:
As you change these code, I suggest to return all these as an error, and print log with errors in `InitializeExecutableWorkers()`.

Reply:
I've changed this to make the behavior for all workers the same. Logically, this is not an error, but a valid use case: a user intentionally disabled the specific (or all) media processing worker. Then the worker got disabled during the init stage, following the user setting. Why should the log contain a record, saying that the error occurred during the worker's init process: worker disabled...?
I'm trying to throw errors only in cases, they are real errors, as according to the `community-support` channel, users take the log output too seriously, complaining even on the health check records with an `unauthorized` word and the missing `.env` file at the app start inside Docker container - if there would be an error about their own setting, that would make a lot of false-complains (I think)

Reply:
It's OK to keep it as what it is now. I'd like to refactor the log sometime in the future.

Reply:
Yeah, the logging in the Photoview globally should be enhanced)